src/custom/state/swap/trade.test.ts

```diff
@@ -0,0 +1,112 @@
+import { parseUnits } from '@ethersproject/units'
+import { getCanonicalMarket } from '@src/custom/utils/misc'
+import { PriceQuoteParams } from '@src/custom/utils/operator'
+import { basisPointsToPercent } from '@src/utils'
+import { ChainId, Fraction, JSBI, Pair, Route, Token, TokenAmount, Trade, TradeType, WETH } from '@uniswap/sdk'
+
+async function mockGetPriceQuote(params: PriceQuoteParams & { mockedPriceOut: string }) {
+  const { quoteToken, mockedPriceOut } = params
+  return Promise.resolve({
+    amount: mockedPriceOut,
+    token: quoteToken
+  })
+}
+
+const WETH_MAINNET = new Token(ChainId.MAINNET, WETH[1].address, 18)
+const DAI_MAINNET = new Token(ChainId.MAINNET, '0x6b175474e89094c44da98b954eedeac495271d0f', 18)
+
+const PAIR_WETH_DAI = new Pair(
+  new TokenAmount(WETH_MAINNET, JSBI.BigInt(1000000000)),
+  new TokenAmount(DAI_MAINNET, JSBI.BigInt(2000000000))
+)
+
+describe('Swap PRICE Quote test', () => {
+  describe('ExactIn: WETH/DAI_MAINNET', () => {
```

Comment:
MAINNET is not relevant info in this test. I would just name it something like: Sell 1 WETH for DAI at 4273 DAI since it's what you do for all tests.